### PR TITLE
Sg global config

### PR DESCRIPTION
**Pull request by badaz**
Everything is in the README.md

### CODE REVIEW

File: src/build.ts [hunk 1]
```diff
@@ -21,20 +22,21 @@ function getModuleNameForPath(path: string): string {
...
-  // a command override is found in the package
-  if (packageJson['watch-module'] && packageJson['watch-module']['command']) {
-    return packageJson['watch-module']['command'];
-  }
+  const moduleConfig = getModuleConfigEntry(path);
+  const command = moduleConfig.command;
 
   // no command override found
-  const yarnOrNpm = hasYarn(path) ? 'yarn' : 'npm';
+  if (command && packageJson['scripts'] && packageJson['scripts'][command]) {
```
Comment (jdeniau) [inline src/build.ts:34]: according to the readme, the command will be `yarn run build:prod` here, so the "real" command will be `yarn run yarn run build:prod`.

I think if a command is defined as configuration, we should not assume that it's a yarn / npm command. You might want to just do shell

File: src/config-utils.ts [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+  return srcPaths;
+}
+
+export function createDefaultConfig(): void {
+  const configPath = getConfigPath();
+  if (!configPath) {
+    return;
+  }
+
+  const configFilePath = `${configPath}/${CONFIG_FILE_NAME}`;
```
Comment (jdeniau) [inline src/config-utils.ts:150]: Apparently, the `.config` is filled with directory, so `~/.config/watch-module/config.json` seems more relevant, no ?

File: src/default-config.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (jdeniau) [inline src/default-config.json:1]: I don't really see the point of this file. Won't it be a huge risk of breaking changes if we want to upgrade the default configuration (or even add a new entry ?)

\+ the "command" cause a problem relative to my previous comment

File: src/config-utils.ts (hunk 2 above)
Comment (badaz) [reply src/config-utils.ts:150]: Yep

File: src/default-config.json (hunk 3 above)
Comment (badaz) [reply src/default-config.json:1]: Good point, so, an empty file then? Or a file with an empty json object ?

File: src/build.ts (hunk 1 above)
Comment (badaz) [reply src/build.ts:34]: Ok, even better

File: src/default-config.json (hunk 3 above)
Comment (badaz) [reply src/default-config.json:1]: Ok, I'm still not used to github comments, I resolved a conversation although it was not finished. I agree that allowing any command would be best

**Review by jdeniau — APPROVED** [review]

**Pull request merged**